File: FrontEnd/src/app/models/Quiz.ts
import { Question } from "./Question";
import { Questionnaire } from "./Questionnaire";
import { QCM } from "./qcm";

export interface Quiz {
  quizId: number; // Le point d'interrogation indique que ce champ est optionnel
  titre: string;
  niveauId: number; // Changez le nom de `niveau` à `niveauId`
  nombreQCM: number;
  questionnaireNiveau: number;
  nombreQuestionnaires: number;
  dureeMinutes: number;
  description: string;
  questionnaireType: string;
  dateCreation?: string; // Optionnel
  qcMs?: any[]; // Optionnel, changez le type selon vos besoins
  questionnaires?: any[]; // Optionnel, changez le type selon vos besoins
}


  
  export interface QCMQuestion {
    option: any;
    quizId: number;
    qcmQuestionId: number;
    qcmId: number;
    qcm: QCM;
    questionId: number;
    question: Question[];
  }
  

File: BackEnd/AndexTech_Talent_Hub_Bibliothéque/Models/Quiz.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace AndexTech_Talent_Hub_Bibliotheque.Models
{
    public class Quiz
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int QuizId { get; set; }
        public string Titre { get; set; }
        public int DureeMinutes { get; set; }
        public DateTime DateCreation { get; set;}
        public string Description { get; set; }
        public List<QCM> QCMs { get; set; }
        public List<Questionnaire> Questionnaires { get; set; }
        public int NiveauId { get; set; }


    }

}
